Source/Core/VideoBackends/Vulkan/Renderer.cpp

```diff
@@ -640,6 +640,23 @@ void Renderer::SwapImpl(AbstractTexture* texture, const EFBRectangle& xfb_region
   TextureCache::GetInstance()->Cleanup(frameCount);
 }
 
+void Renderer::OnSwapInvalid()
+{
+  // End the current render pass.
```

Comment:
This can all be simplified to Util::ExecuteCurrentCommandsAndRestoreState(true, false).

Reply:
`Util::ExecuteCurrentCommandsAndRestoreState` finally call `g_command_buffer_mgr->SubmitCommandBuffer`, so it can be  simpled to `g_command_buffer_mgr->SubmitCommandBuffer(true);`, and it also submit command buffer on worker thread.

Reply:
you're right. it's more clean.

Reply:
Are you sure this is correct now? BeginFrame() will result in duplicate calls to the state manager, as well as CommandBufferManager::ActivateCommandBuffer(). Checking these sorts of changes against the validation layer is usually a good idea.

(yes, I know the CommandBufferManager is confusing, refactoring it has been on my todo list for a while)